### PR TITLE
echosign module

### PR DESCRIPTION
**Pull request by z0d14c**
Added a module for Adobe Echosign Service.

Seems to be working for me (getting tokens).

### CODE REVIEW

**Comment by logical-and** [comment]
Hey. :)
Can you please open pull request at [updated library](https://github.com/logical-and/php-oauth), so I will be able to merge your changes.

File: src/OAuth/OAuth2/Service/Echosign.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        CredentialsInterface $credentials,
+        ClientInterface $httpClient,
+        TokenStorageInterface $storage,
+        $scopes = array(),
+        UriInterface $baseApiUri = null
+    ) {
+        parent::__construct($credentials, $httpClient, $storage, $scopes, $baseApiUri);
+
+        if (null === $baseApiUri) {
+            $this->baseApiUri = new Uri('https://api.echosign.com/api/rest/v3 ');
```
Comment (elliotchance) [inline src/OAuth/OAuth2/Service/Echosign.php]: I'm sure you didn't mean the space at the end?

File: src/OAuth/OAuth2/Service/Echosign.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        if (null === $data || !is_array($data)) {
+            throw new TokenResponseException('Unable to parse response.');
+        } elseif (isset($data['error'])) {
+            throw new TokenResponseException('Error in retrieving token: "' . $data['error'] . '"');
+        }
+
+        $token = new StdOAuth2Token();
+        $token->setAccessToken($data['access_token']);
+        // tokens evidently never expire... TODO: not sure about this
```
Comment (elliotchance) [inline src/OAuth/OAuth2/Service/Echosign.php]: Have you confirmed this is now the case?

**Comment by elliotchance** [comment]
@z0d14c please add some unit tests.

**Pull request closed without merging**